src/python/aqueduct_executor/operators/utils/utils.py

```diff
@@ -89,6 +90,53 @@ def read_artifacts(
     return inputs, artifact_types, serialization_types
 
 
+def read_artifacts_spark(
```

Comment:
`read_artifacts_spark` and `write_artifact_spark` are the main functions to review. Currently the sample method is df.limit(100). Sampling with spark [dataframes requires a fraction](https://spark.apache.org/docs/3.1.2/api/python/reference/api/pyspark.sql.DataFrame.sample.html), and can still result in an overlarge size if the original DF is large enough. I'd like to hear thoughts/suggestions on this approach